hudi-common/src/main/java/org/apache/hudi/common/config/DefaultHoodieConfig.java

```diff
@@ -44,6 +56,53 @@ public static void setDefaultOnCondition(Properties props, boolean condition, De
     }
   }
 
+  public static void setDefaultValue(Properties props, ConfigOption configOption) {
+    if (!contains(props, configOption)) {
+      Option<String> inferValue = Option.empty();
+      if (configOption.getInferFunc() != null) {
+        inferValue = (Option<String>) configOption.getInferFunc().apply(props);
+      }
+      props.setProperty(configOption.key(), inferValue.isPresent() ? inferValue.get() : configOption.defaultValue().toString());
+    }
+  }
+
+  public static void setDefaultValue(Map<String, String> props, ConfigOption configOption) {
+    if (!contains(props, configOption)) {
+      Option<String> inferValue = Option.empty();
+      if (configOption.getInferFunc() != null) {
+        inferValue = (Option<String>) configOption.getInferFunc().apply(props);
+      }
+      props.put(configOption.key(), inferValue.isPresent() ? inferValue.get() : configOption.defaultValue().toString());
+    }
+  }
+
+  public static boolean contains(Map props, ConfigOption configOption) {
+    if (props.containsKey(configOption.key())) {
+      return true;
+    }
+    return Arrays.stream(configOption.getDeprecatedNames()).anyMatch(props::containsKey);
+  }
+
+  public static Option<Object> getRawValue(Properties props, ConfigOption configOption) {
+    if (props.containsKey(configOption.key())) {
+      return Option.of(props.get(configOption.key()));
+    }
+    for (String deprecateName : configOption.getDeprecatedNames()) {
+      if (props.containsKey(deprecateName)) {
+        return Option.of(props.get(deprecateName));
+      }
+    }
```

Comment:
- Shouldn't you be using Option.ofNullable() to avoid null pointer exceptions here ?
- nit : deprecateName => deprecatedName